dev/src/types.ts

```diff
@@ -92,6 +92,77 @@ export type UnaryMethod<Req, Resp> = (
 // tslint:disable-next-line:no-any
 export type RBTree = any;
 
+/**
+ * Converter used by `withConverter()` to transform user objects of type T
+ * into Firestore data.
+ *
+ * Using the converter allows you to specify generic type arguments when
+ * storing and retrieving objects from Firestore.
+ *
+ * @example
+ * ```typescript
+ * class Post {
+ *   constructor(readonly title: string, readonly author: string) {}
+ *
+ *   toString(): string {
+ *     return this.title + ', by ' + this.author;
+ *   }
+ * }
+ *
+ * const postConverter = {
+ *   toFirestore(post: Post): FirebaseFirestore.DocumentData {
+ *     return {title: post.title, author: post.author};
+ *   },
+ *   fromFirestore(
+ *     data: FirebaseFirestore.DocumentData
+ *   ): Post {
+ *     return new Post(data.title, data.author);
+ *   }
+ * };
+ *
+ * const postSnap = await Firestore()
+ *   .collection('posts')
+ *   .withConverter(postConverter)
+ *   .doc().get();
+ * const post = postSnap.data();
+ * if (post !== undefined) {
+ *   post.title; // string
+ *   post.toString(); // Should be defined
+ *   post.someNonExistentProperty; // TS error
+ * }
+ * ```
+ */
+export interface FirestoreDataConverter<T> {
```

Comment:
Since this is a type, this does not actually show up in our JSDocs. Can you move this example to the `withConverter()` methods? Without this interface or this example, these methods are not very explanatory. 

You can look at the docs output yourself by running `npm run docs`. This spits the docs into ./docs/